src/js/utils/background.js

```diff
@@ -159,8 +165,17 @@ export const backgroundAndTextColors = (backgroundArg, textArg, theme) => {
   } else {
     backgroundColor = normalizeBackgroundColor(background, theme);
     const shade = darkContext(backgroundColor, theme);
+    let transparent;
+
+    if (backgroundColor && canExtractRGBArray(backgroundColor)) {
+      const colorArray = getRGBArray(backgroundColor);
+      // check if the alpha value is less than 0.5
+      if (colorArray[3] < 0.5) transparent = true;
+    }
     if (shade) {
       textColor = normalizeColor(text[shade] || text, theme, shade === 'dark');
+    } else if (transparent && text) {
+      textColor = normalizeColor(text, theme);
```

Comment:
The parent container has `inherit` for text color. Not sure if it's preferred to just use `inherit` here as opposed to recalculating as value?
```suggestion
      textColor = 'inherit';
```